routes/scan.js

```diff
@@ -3,25 +3,30 @@
 const express = require("express");
 const bodyParser = require("body-parser");
 
-const HIBP = require("../hibp");
+const sha1 = require("../sha1-utils");
 
+const HIBP = require("../hibp");
 
 const router = express.Router();
 const urlEncodedParser = bodyParser.urlencoded({ extended: false });
 
-
+const blankStringSha1 = sha1("");
 
 router.post("/", urlEncodedParser, async (req, res) => {
   const emailHash = req.body.emailHash;
+
+  if (!emailHash || emailHash === blankStringSha1) {
+    res.redirect("/");
+    return;
+  }
+
   let foundBreaches = [];
 
   if (!req.session.scanResults) {
     req.session.scanResults = {};
   }
 
-  if (emailHash) {
-    foundBreaches = await HIBP.getBreachesForEmail(emailHash);
-  }
+  foundBreaches = await HIBP.getBreachesForEmail(emailHash);
```

Comment:
Should we just do a `const foundBreaches = ...` here and nuke L23 above? Not sure why we’d set to empty array and then overwrite here.